sql/core/src/test/scala/org/apache/spark/sql/execution/vectorized/ColumnarBatchSuite.scala

```diff
@@ -326,6 +329,80 @@ class ColumnarBatchSuite extends SparkFunSuite {
     }}
   }
 
+  test("CachedBatch primitive Apis") {
```

Comment:
let's follow the arrow PR and have a new test suite: https://github.com/apache/spark/pull/18680/files#diff-723b0ca0368980a792792bb010da86c9

We should also have an individual test cases for each data type.